### PR TITLE
nc-news-11

### PR DESCRIPTION
**Pull request by Harrywg**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -75,6 +75,133 @@ describe("/api/articles", () => {
...
+                article_img_url: expect.any(String),
+              })
+            );
+          });
+        });
+    });
+    test("200 + ?sort_by ignores invalid queries", () => {
+      return request(app)
+        .get("/api/articles?sort_by=bananas")
+        .expect(200)
```
Comment (johnhutcheon) [inline __tests__/app.test.js]: We should be expecting a 400 here as it's an invalid sort_by query 👍

File: __tests__/app.test.js [hunk 2]
```diff
@@ -75,6 +75,133 @@ describe("/api/articles", () => {
...
+                article_img_url: expect.any(String),
+              })
+            );
+          });
+        });
+    });
+    test("200 + ?order ignores invalid queries", () => {
+      return request(app)
+        .get("/api/articles?order=bananas")
+        .expect(200)
```
Comment (johnhutcheon) [inline __tests__/app.test.js]: Again, this should be a 400

File: __tests__/app.test.js [hunk 3]
```diff
@@ -75,6 +75,133 @@ describe("/api/articles", () => {
...
+              votes: expect.any(Number),
+              article_img_url: expect.any(String),
+            })
+          );
+        });
+    });
+    test("200 + ?topic returns no articles if can't find given topic", () => {
+      return request(app)
+        .get("/api/articles?topic=banana")
+        .expect(200)
```
Comment (johnhutcheon) [inline __tests__/app.test.js:175]: This should be a 404

File: __tests__/app.test.js [hunk 4]
```diff
@@ -75,6 +75,133 @@ describe("/api/articles", () => {
...
+        .then(({ body }) => {
+          const articles = body.articles;
+          expect(articles.length).toBe(0);
+        });
+    });
```
Comment (johnhutcheon) [inline __tests__/app.test.js:179]: Could we add another test in here please? What would we expect to happen if we query a valid topic, but it has no articles. E.g. "paper" in the test data.

File: error-handler.js [hunk 5]
```diff
@@ -1,5 +1,6 @@
 const express = require("express");
 exports.errorHandler = (err, req, res, next) => {
+  console.log(err);
```
Comment (johnhutcheon) [inline error-handler.js]: make sure you remove any console logs and comments before pushing code 👍

**Review by johnhutcheon — CHANGES_REQUESTED** [review]
Hi Harry, just need a bit more work on your testing please if you can see my comments. The model will need to be amended too. See how you get on, use the notes, and if you get stuck give us an nchelp 🙂

File: models/articles-model.js [hunk 6]
```diff
@@ -11,8 +11,37 @@ exports.selectArticlesById = (params) => {
...
+    case "created_at":
+    case "votes":
+    case "article_img_url":
+      break;
+    case undefined:
+      sort_by = "created_at";
+      break;
+    default:
+      return Promise.reject({ code: 400, msg: "Bad Request", custom: true });
+  }
```
Comment (Slyose) [inline models/articles-model.js:32]: A more concise way to solve this is by placing the valid sort-bys in an array, then using . includes to check if the sort_by that has been passed to the function is valid or not.

That would make your function more readable and easily understandable, and is a more expected way of handling this.

File: __tests__/app.test.js (hunk 4 above)
Comment (Slyose) [reply __tests__/app.test.js:171]: Not seen a response to this feedback yet!

**Review by Slyose — COMMENTED** [review]
Good work so far! I left an optional comment that could help improve your model. The last thing this needs is a test metioned in John's feedback - 

"What would we expect to happen if we query a valid topic, but it has no articles. E.g. "paper" in the test data."

Try test the url - `/api/articles?topic=paper`!

**Review by Slyose — COMMENTED** [review]
About your comments, I'd definitely call using arrays a 'better' solution because of it being concise / readable and easier to make changes to, but exactly it's not the end of the world to use a switch statement

And in terms of the last test, it'd want to be a 200. This means it wouldn't need its own extra handling. As long as your code doesn't error out and sends a 200 with an empty array, you're in the clear

**Review by Slyose — APPROVED** [review]
Nice one! Go ahead and hit the big green button ✅

**Pull request merged**